Content.Shared/Stacks/SharedStackSystem.cs

```diff
@@ -190,6 +196,35 @@ public bool Use(EntityUid uid, int amount, StackComponent? stack = null)
             return true;
         }
 
+        /// <summary>
+        /// Tries to merge a stack into any of the stacks it is touching.
+        /// </summary>
+        /// <returns>Whether or not it was successfully merged into another stack</returns>
+        public bool TryMergeToContacts(EntityUid uid, StackComponent? stack = null, TransformComponent? xform = null)
+        {
+            if (!Resolve(uid, ref stack, ref xform, false))
+                return false;
+
+            var map = xform.MapID;
+            var bounds = _physics.GetWorldAABB(uid);
+            var intersecting = _entityLookup.GetComponentsIntersecting<StackComponent>(map, bounds,
+                LookupFlags.Approximate | LookupFlags.Uncontained);
```

Comment:
Does uncontained include static? You can probably just use dynamic + sundries for this.